### PR TITLE
add extrap tutorial

### PR DESCRIPTION
**Pull request by vanessalama09**
The goal of the PR is to add tutorial material to use extra with thicket. This includes a tutorial notebook and a few cali files used in the notebook.

### CODE REVIEW

File: notebooks/extrap-with-metadata-aggregated.ipynb [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+   "source": [
+    "import sys\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import pandas as pd\n",
+    "from IPython.display import display\n",
+    "from IPython.display import HTML\n",
+    "\n",
+    "import thicket as th\n",
+    "\n",
```
Comment (slabasan) [inline notebooks/extrap-with-metadata-aggregated.ipynb]: No new line here, both are local packages
```
import thicket as th
from thicket.model_extrap import Modeling
```

File: notebooks/extrap-with-metadata-aggregated.ipynb [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    "t_ens.metadata[\"jobsize\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. More Information on a Function\n",
+    "***\n",
+    "You can use the help() method within Python to see the information for a given object. You can do this by typing help(object). \n",
```
Comment (slabasan) [inline notebooks/extrap-with-metadata-aggregated.ipynb]: `help()`, `help(object)1

File: notebooks/extrap-with-metadata-aggregated.ipynb [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    "mdl.produce_models()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Models Dataframe\n",
+    "\n",
+    "Model hypothesis functions are stored in the thicket's statsframe."
```
Comment (slabasan) [inline notebooks/extrap-with-metadata-aggregated.ipynb]: statsframe --> aggregated statistics table

File: notebooks/extrap-with-metadata-aggregated.ipynb [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Create Models\n",
+    "\n",
+    "First, we construct the `Modeling` object by passing all the relevant data to it. We provide `jobsize` as the `param_name` argument so the model will grab this column from the metadata table to use as our parameter. We also sub-select some metrics, since this dataset has a lot of metrics (otherwise the modeling will take a long time to do all metrics).\n",
+    "\n",
+    "Then, we call `produce_models` on that object (it's unecessary to provide an aggregation function since the data is already aggregated.).\n",
+    "\n",
+    "**NOTE:** For this example, you can view all the metric columns by adding a new cell and running: `t_ens.dataframe.columns`. "
```
Comment (slabasan) [inline notebooks/extrap-with-metadata-aggregated.ipynb]: Should this be `t_ens.performance_cols`?

File: notebooks/extrap-with-metadata-aggregated.ipynb [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+   "source": [
+    "# Thicket and ExtraP: Thicket Tutorial\n",
+    "\n",
+    "Thicket is a python-based toolkit for Exploratory Data Analysis (EDA) of parallel performance data that enables performance optimization and understanding of applications’ performance on supercomputers. It bridges the performance tool gap between being able to consider only a single instance of a simulation run (e.g., single platform, single measurement tool, or single scale) and finding actionable insights in multi-dimensional, multi-scale, multi-architecture, and multi-tool performance datasets.\n",
+    "\n",
+    "#### NOTE: An interactive version of this notebook is available in the Binder environment.\n",
+    "\n",
+    "[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/llnl/thicket-tutorial/develop)\n",
+    "\n",
+    "# Hatchet Modeling Example\n",
```
Comment (slabasan) [inline notebooks/extrap-with-metadata-aggregated.ipynb]: Update references from Hatchet to Thicket

File: requirements.txt [hunk 6]
```diff
@@ -1,2 +1,3 @@
-llnl-thicket
-ipython
\ No newline at end of file
+llnl-thicket[]
```
Comment (slabasan) [inline requirements.txt]: do you need blank brackets?

File: notebooks/extrap-with-metadata-aggregated.ipynb [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    "model_obj.eval(600)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Displaying the model:\n",
+    "\n",
+    "It returns a _figure_ and an _axis_ objects. The axis object can be used to adjust the plot, i.e., change labels. The display() function requires an input for RSS (bool), that determines whether to display Extra-P RSS on the plot."
```
Comment (slabasan) [inline notebooks/extrap-with-metadata-aggregated.ipynb]: use tick marks around `display()` and `RSS`

File: notebooks/extrap-with-metadata-aggregated.ipynb [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "with pd.option_context('display.max_colwidth', 1):\n",
```
Comment (slabasan) [inline notebooks/extrap-with-metadata-aggregated.ipynb]: use double quotes in code: `with pd.option_context("display.max_colwidth", 1):\n`

File: notebooks/extrap-with-metadata-aggregated.ipynb [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+    "    display(HTML(mdl.to_html()))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 6. Query Specific Model\n",
+    "\n",
+    "The 1st node `{'name': 'MPI_Allreduce', 'type': 'function'}`, has an interesting graph so we want to retrieve its model. This can be achieved by indexing the `models_df` DataFrame for our chosen node for the metric `Avg time/rank_extrap-model`."
```
Comment (slabasan) [inline notebooks/extrap-with-metadata-aggregated.ipynb]: {"name": "MPI_Allreduce", "type": "function"}

File: requirements.txt (hunk 6 above)
Comment (vanessalama09) [reply requirements.txt]: no, it's not needed, that was an accident!

File: notebooks/extrap-with-metadata-aggregated.ipynb (hunk 4 above)
Comment (vanessalama09) [reply notebooks/extrap-with-metadata-aggregated.ipynb]: I personally used t_ens.dataframe.columns to check for the metric columns

**Pull request merged**